src/libstd/sync/mutex.rs

```diff
@@ -252,6 +253,18 @@ impl<T: Send + 'static> Drop for Mutex<T> {
     }
 }
 
+#[stable(feature = "rust1", since = "1.0.0")]
+impl<T: fmt::Debug + Send + 'static> fmt::Debug for Mutex<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self.try_lock() {
+            Ok(guard) => write!(f, "Mutex {{ data: {:?} }}", *guard),
+            Err(TryLockError::Poisoned(err)) =>
+                write!(f, "Mutex {{ data: Poisoned({:?}) }}", **err.get_ref()),
+            Err(_) => write!(f, "Mutex {{ <locked> }}")
```

Comment:
Generally it's more robust to use exhaustive matching wherever possible instead of a catch-all. In this case there's only one more variant so you may wish to explicitly match it.

Also, whenever a match arm goes beyond one line we put braces around it as well.